js/utilities.js

```diff
@@ -0,0 +1,96 @@
+/* global someFunction jQuery:true */
+
+/*
+ * Fuel UX Utilities
+ * https://github.com/ExactTarget/fuelux
+ *
+ * Copyright (c) 2016 ExactTarget
+ * Licensed under the BSD New license.
+ */
+
+// -- BEGIN UMD WRAPPER PREFACE --
+
+// For more information on UMD visit:
+// https://github.com/umdjs/umd/blob/master/templates/jqueryPlugin.js
+
+// Uses CommonJS, AMD or browser globals to create a jQuery plugin.
+
+(function umdFactory (factory) {
+	if (typeof define === 'function' && define.amd) {
+		// AMD. Register as an anonymous module.
+		define(['jquery'], factory);
+	} else if (typeof module === 'object' && module.exports) {
+		// Node/CommonJS
+		module.exports = function exportsFunction ( root, jQuery ) {
+			var jQueryReturn;
+
+			if ( typeof jQuery === 'undefined' ) {
+				// require('jQuery') returns a factory that requires window to
+				// build a jQuery instance, we normalize how we use modules
+				// that require this pattern but the window provided is a noop
+				// if it's defined (how jquery works)
+				if ( typeof window !== 'undefined' ) {
+					jQueryReturn = require('jquery');
+				} else {
+					jQueryReturn = require('jquery')(root);
+				}
+			}
+			factory(jQueryReturn);
+			return jQueryReturn;
+		};
+	} else {
+		// Use browser globals if AMD is not present
+		factory(jQuery);
+	}
+}(function makePlugin ($) {
+	// -- END UMD WRAPPER PREFACE --
+	// -- BEGIN MODULE CODE HERE --
+
+	var BACKSPACE_KEYCODE = 8;
+	var COMMA_KEYCODE = 188;// `,` & `<`
+	var DELETE_KEYCODE = 46;
+	var DOWN_ARROW_KEYCODE = 40;
+	var ENTER_KEYCODE = 13;
+	var TAB_KEYCODE = 9;
+	var UP_ARROW_KEYCODE = 38;
+
+	var isShiftHeld = function isShiftHeld(e) { return e.shiftKey === true; };
+	var isBackspaceKey = function isBackspaceKey(e) { return e.keyCode === BACKSPACE_KEYCODE; };
+	var isDeleteKey = function isDeleteKey(e) { return e.keyCode === DELETE_KEYCODE; };
+	var isTabKey = function isTabKey(e) { return e.keyCode === TAB_KEYCODE; };
+	var isUpArrow = function isUpArrow(e) { return e.keyCode === UP_ARROW_KEYCODE; };
+	var isDownArrow = function isDownArrow(e) { return e.keyCode === DOWN_ARROW_KEYCODE; };
```

Comment:
any reason to not dry this up
https://github.com/ExactTarget/fuelux/pull/1856#discussion_r75308689